src/utils/file.cpp

```diff
@@ -210,6 +241,14 @@ namespace file_util {
     return stat(filename.c_str(), &buffer) == 0 && S_ISFIFO(buffer.st_mode);
   }
 
+  /*
+   * Checks if the (potentionally non-canonical) path is absolute
+   */
+  bool is_absolute(const string& filename) {
+    // this is sufficient for Unix-like systems
+    return !filename.empty() && (filename[0] == '/' || filename[0] == '~');
```

Comment:
Wouldn't this say `$HOME/.config/polybar/config` is relative?

Reply:
Good catch. I made the `expand_relative_to` function expand the path before checking if it's absolute.